app/controllers/items_controller.rb

```diff
@@ -24,31 +24,16 @@ def create
 
   def update
     item.update!(item_params)
-    respond_to do |format|
-      format.json do
-        render json: item.as_json(only: [:id, :desc, :completed])
-      end
-
-      format.html do
-        flash[:notice] = "Item has been updated"
-        redirect_to workspace
-      end
-    end
+
+    flash.now[:notice] = "Item has been updated"
+    render partial: "update", locals: {item}, content_type: "text/vnd.turbo-stream.html"
```

Comment:
```suggestion
    respond_to do |format|
      format.turbo_stream do
        render partial: "update", locals: {item}
      end
      format.html do
        flash[:notice] = "Item has been updated"
        redirect_to workspace
      end
    end
```
Untested! But I think this addresses the <em><q>I found that Turbo expects HTTP response to have the `text/vnd.turbo-stream.html` content type in order to activate stream elements</q></em> confusion in [your (great!) article](https://evilmartians.com/chronicles/hotwire-reactive-rails-with-no-javascript). See the [`Turbo::StreamsHelper` documentation](https://github.com/hotwired/turbo-rails/blob/e551a53c8f48a331e50c886ba89b62511fe34a9c/app/helpers/turbo/streams_helper.rb#L5-L26) for more.

Reply:
I rewrote this part a bit: https://github.com/anycable/anycable_rails_demo/pull/16/commits/860c706d76acd911d20bc5a9f3b2447a92ce561a

Reply:
Sweet! Maybe update your blog post too?

Reply:
Already updated)

Reply:
I was thinking of this section in particular:
<img src="https://user-images.githubusercontent.com/5355/114583123-4510a400-9c4f-11eb-90d0-9e398795f222.png" width="500">
The preferred way to respond with Turbo Streams in Rails is with implicit `.turbo_stream.erb` templates (e.g. 860c706)  and/or `format.turbo_stream { … }` (in `respond_to`) blocks. They set the custom response content type automatically.

Reply:
Will it work without `respond_to`? I mean: `render turbo_stream: "update", locals: ...`.